app/models/ability.rb

```diff
@@ -211,8 +211,10 @@ def signed_in_with_cfp_role(user)
       (Conference.with_role(:cfp, user).pluck(:id).include? role.resource_id)
     end
 
+    can [:index, :revert_object, :revert_attribute], PaperTrail::Version, item_type: 'Event', conference_id: conf_ids_for_cfp
     can [:index, :revert_object, :revert_attribute], PaperTrail::Version do |version|
-      %w(Event Commercial).include?(version.item_type) && (conf_ids_for_cfp.include? version.conference_id)
+      version.item_type == 'Commercial' && conf_ids_for_cfp.include?(version.conference_id) &&
+      (version.object.to_s.include?('Event') || version.object_changes.to_s.include?('Event'))
```

Comment:
Querying for versions of event commercials is tricky.
For all PaperTrail::Version objects,
 -  object field has a representation of the object before the changes in that version.
    So object is nil for 'create' type versions
 -  object_changes field has representation of the changes in the object from its previous state.So 'create' type versions' object_changes field will have its commercialable_type in it.
Hence the two checks here.